cli/src/diff_util.rs

```diff
@@ -677,19 +677,48 @@ fn show_unified_diff_hunks(
     Ok(())
 }
 
+fn materialized_diff_stream<'a>(
+    store: &'a Store,
+    tree_diff: TreeDiffStream<'a>,
+) -> Pin<
+    Box<
+        dyn Stream<
+                Item = (
+                    RepoPathBuf,
+                    BackendResult<(MaterializedTreeValue, MaterializedTreeValue)>,
+                ),
+            > + 'a,
+    >,
+> {
```

Comment:
Better to return `impl Stream`? The complete type looks super complex.

Appears that `impl Stream<Item = (..)> + 'a` works, and `Box::pin()` can be removed.

Reply:
Good idea.